contribs/application/src/main/java/org/matsim/freightDemandGeneration/CarrierReaderFromCSV.java

```diff
@@ -463,20 +457,21 @@ static void createNewCarrierAndAddVehilceTypes(Scenario scenario,
 				carriers.addCarrier(thisCarrier);
 			}
 			if (singleNewCarrier.getVehicleDepots() == null)
-				singleNewCarrier.setVehicleDepots(new String[] {});
-			while (singleNewCarrier.getVehicleDepots().length < singleNewCarrier.getNumberOfDepotsPerType()) {
-				Random rand = new Random();
+				singleNewCarrier.setVehicleDepots(new ArrayList<String>());
+			Random rand = new Random(4711);

```

Comment:
Maybe initialise `rand` with a seed like `singleNewCarrier.name.hashCode()`. Otherwise you will always get the same link sequence for all carriers, which may not be desired if you want to have them more spread. `singleNewCarrier.name.hashCode()` makes it also reproducible

Reply:
Thanks for this idea. I will change it